diff.py

```diff
@@ -1609,6 +1674,62 @@ def _normalize_adrp_differences(self, mnemonic: str, row: str) -> str:
         return row
 
 
+class AsmProcessorI686(AsmProcessor):
+    def process_reloc(self, row: str, prev: str) -> Tuple[str, Optional[str]]:
+        arch = self.config.arch
+        repl = row.split()[-1]
+        args_string = ""
+        reverse = False
+        mnemonic, args = prev.split(maxsplit=1)
+
+        comma = args.find(",")
+        open_bracket = args.find("(")
+
+        # Register relative addressed target e.g.
+        # R_386_GOTOFF        x
+        # movl      $0x1,0x0(%eax)
+        if comma != -1:
+            if open_bracket != -1 and open_bracket > comma:
+                reverse = True
+                args_string = args[: comma + 1]
+            else:
+                args_string = args[comma:]
+
+        if "R_386_NONE" in row:
+            pass
+        elif "R_386_32" in row:
+            pass
+        elif "R_386_PC32" in row:
+            pass
+        elif "R_386_16" in row:
+            pass
+        elif "R_386_PC16" in row:
+            pass
+        elif "R_386_8" in row:
+            pass
+        elif "R_386_PC8" in row:
+            pass
+        elif "R_386_GOT32" in row:
+            repl = f"%got({repl})"
+        elif "R_386_PLT32" in row:
+            repl = f"%plt({repl})"
+        elif "R_386_RELATIVE" in row:
+            repl = f"%rel({repl})"
+        elif "R_386_GOTOFF" in row:
+            repl = f"%got({repl})"
+        elif "R_386_GOTPC" in row:
+            repl = f"%got({repl})"
+        elif "R_386_32PLT" in row:
+            repl = f"%plt({repl})"
+        else:
+            assert False, f"unknown relocation type '{row}' for line '{prev}'"
+
+        if reverse:
```

Comment:
rather than a `reverse` variable I think it'd be cleaner to have `before` and `after`, similar to reloc code for most other arches